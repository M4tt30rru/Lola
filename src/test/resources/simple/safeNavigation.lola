##Import "std.lola"
##Find ##Expression(_exp1) ?. ##Either ##Identifier(_exp2) ##or ##Literal(_exp2)
 ##replace Optional.of(##(_exp1)).map(##(_exp2)).orElse(null)

f() ?. d1 ?. d2
