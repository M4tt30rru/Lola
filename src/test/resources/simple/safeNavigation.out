Optional.of(Optional.of(f()).map(d1).orElse(null)).map(d2).orElse(null)
